### DISCUSSION TITLE
Retry static exports

### DISCUSSION THREAD

**Discussion by dmitrizzle** [vercel/next.js#11796] [Help]
Is there a way to have Next retry building a static export? In my case, the build requires fetching data from remote API, which occasionally fails. I have 400+ paths to render so the chances of this happening is quite high.

```
[  ==] Exporting (422/456)
Error occurred prerendering page "640/page" https://err.sh/zeit/next.js/prerender-error: { FetchError: invalid json response body at https://calculate.qxmd.com/fetch-items reason: Unexpected token < in JSON at position 0
    at /app/node_modules/node-fetch/lib/index.js:272:32
    at process._tickCallback (internal/process/next_tick.js:68:7)
  message:
   'invalid json response body at https://api.com/xxx reason: Unexpected token < in JSON at position 0',
  type: 'invalid-json' }
```
Thank you.

**Comment by ingoclaro** [comment]
for the time being you could try caching the results of the api call into your local filesystem, then when rebuilding check your cache first and use that data instead of doing the api call.
Not ideal, but it would unblock you...

**Comment by Timer** [comment]
It'd probably be better to use something like [`fetch-retry`](https://github.com/zeit/fetch-retry) than relying on Next.js to do this.

**Comment by dmitrizzle** [comment]
I've tried the retry logic, however, the issue persists. The 500x errors are no longer there, however, I get `socket hang up` which, if I understand correctly is due to having too many concurrent requests going out from Node into remote API. Can you recommend how these may be spaced out?

My total paths map has 400+ pages. The additional complexity is in the fact that I have to run two servers out of the same code base: static (the one I'm having issues rendering with) and dynamic (which has no problems but should not be affected by fixes to the static app).
